provider/resource_harbor_webhook.go

```diff
@@ -0,0 +1,239 @@
+package provider
+
+import (
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/liatrio/terraform-provider-harbor/harbor"
+)
+
+func resourceWebhook() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceWebhookCreate,
+		Read:   resourceWebhookRead,
+		Update: resourceWebhookUpdate,
+		Delete: resourceWebhookDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:         schema.TypeString,
+				Description:  "ID of the project the webhook corresponds to, in the form '/projects/${ID_NUMBER}'",
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile(`^/projects/[0-9]+$`), "validation error: project_id should be of the form '/projects/${ID_NUMBER}'"),
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Description:  "Display name of the webhook.",
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 256),
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Description: "Contains a description of the webhook.",
+				Optional:    true,
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Description: "When true, webhooks are enabled.",
+				Optional:    true,
+				Default:     true,
+			},
+			"event_types": {
+				Type:        schema.TypeSet,
+				Description: "The list of events that will cause this webhook to trigger.",
+				Required:    true,
+				MinItems:    1,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice(
+						[]string{
+							"DELETE_ARTIFACT",
+							"PULL_ARTIFACT",
+							"PUSH_ARTIFACT",
+							"DELETE_CHART",
+							"DOWNLOAD_CHART",
+							"UPLOAD_CHART",
+							"QUOTA_EXCEED",
+							"QUOTA_WARNING",
+							"REPLICATION",
+							"SCANNING_FAILED",
+							"SCANNING_COMPLETED",
+							"TAG_RETENTION",
+						},
+						false,
+					),
+				},
+			},
+			"target": {
```

Comment:
can there be multiple targets for one webhook?  the UI made it seem like there was one, but maybe it isn't doing a good job representing the data model

Reply:
According to the API there can be multiple, I haven't actually tested that it sends payloads correctly, but there's no doubt that that API does accept multiple targets and stores that information.